protostar/compiler/compiled_contract_writer.py

```diff
@@ -0,0 +1,45 @@
+import json
+from pathlib import Path
+
+from starkware.starknet.services.api.contract_class import ContractClass
+
+
+class CompiledContractWriter:
```

Comment:
this could be a plain function

Reply:
In this case, it could. Something feels off when mixing paradigms.

It's easier to find a file that contains a class rather than a list of functions. Naming functions in the same way as files is confusing because a function and a filename use the same casing.

Reply:
You can still define single public module function, and name the module the same way. That's what I have seen in the wild.

> Something feels off when mixing paradigms.

To me, using classes too much in Python/JS is mixing paradigms.

Reply:
> You can still define single public module function, and name the module the same way. That's what I have seen in the wild.

I did it when I used the functional paradigm. Auto import functionality confused the function with the module, if they were named in the same way. Also, I couldn't patch the function, if it was imported from the `__init__.py` (barrel file) from the same reason.

> To me, using classes too much in Python/JS is mixing paradigms.

Functional programming is more common in Python, but it doesn't mean that this is the proper tool for the job.

> -  Object-Oriented Programming is the tool best suited for defining how we cross architectural boundaries with polymorphism and plugins
> - Functional programming is the tool we use to push data to the boundaries of our applications
> - and Structured programming is the tool we use to write algorithms

https://khalilstemmler.com/articles/software-design-architecture/full-stack-software-design/